devito/function.py

```diff
@@ -223,6 +223,7 @@ def _initialize_derivatives(self):
         """
         Dynamically create notational shortcuts for space derivatives.
         """
+        derivatives = ()
```

Comment:
always use a ``list``, not ``tuple``, when you have full control and you're going to change its content. The difference is:
- if you do ``list.append`` then you're modifying the list "in place"
- if you use ``+=`` (no matter whether it's a tuple, or list, or a string ...), then you're constructing a **new** object. 

Although this is practically irrelevant here, it's useful to keep this distinction in mind because in some contexts it really does make a difference

Reply:
WIll change it to list